src/test/regress/expected/multi_insert_select_conflict.out

```diff
@@ -0,0 +1,423 @@
+CREATE SCHEMA on_conflict;
+SET search_path TO on_conflict, public;
```

Comment:
One test suggestion with foreign keys to reference tables:
```SQL
CREATE TABLE test_ref_table (key int PRIMARY KEY);
SELECT create_Reference_table('test_ref_table');
ALTER TABLE target_table ADD CONSTRAINT fkey FOREIGN KEY (col_1) REFERENCES test_ref_table(key) ON DELETE CASCADE;

BEGIN;
 TRUNCATE test_ref_table CASCADE;

 insert into target_table  SELECT col_2, col_1 from source_table_1 ON CONFLICT (col_1) DO UPDATE set col_2 = 55 RETURNING *;
ERROR:  cannot execute parallel COPY on relation "target_table" after DDL command on reference relation "test_ref_table" because there is a foreign key between them and "test_ref_table" has been modified in this transaction
DETAIL:  COPY to a distributed table uses a separate set of connections which will not be able to see the uncommitted changes to the reference table.
HINT:  Perform the COPY in a separate transaction.

```

And,

```SQL
postgres=# BEGIN;
BEGIN
 DELETE FROM test_ref_table ;
DEBUG:  switching to sequential query execution mode
DETAIL:  Reference relation "test_ref_table" is modified, which might lead to data inconsistencies or distributed deadlocks via parallel accesses to hash distributed relations due to foreign keys. Any parallel modification to those hash distributed relations in the same transaction can only be executed in sequential query execution mode
DELETE 3
insert into target_table  SELECT col_2, col_1 from source_table_1 ON CONFLICT (col_1) DO UPDATE set col_2 = 55 RETURNING *;                                                                      DEBUG:  cannot perform distributed INSERT INTO ... SELECT because the partition columns in the source table and subquery do not match
DETAIL:  The target table's partition column should correspond to a partition column in the subquery.
DEBUG:  Collecting INSERT ... SELECT results on coordinator
ERROR:  cannot execute parallel COPY on relation "target_table" after DML command on reference relation "test_ref_table" because there is a foreign key between them and "test_ref_table" has been modified in this transaction
DETAIL:  COPY to a distributed table uses a separate set of connections which will not be able to see the uncommitted changes to the reference table.
HINT:  Perform the COPY in a separate transaction.
```

Reply:
And, some passing tests where we only select from target table which has fkey to reference table, and insert into another table:
```SQL
BEGIN;
 TRUNCATE test_ref_table CASCADE;
 insert into source_table_1  SELECT col_2, col_1 from target_table ON CONFLICT (col_1) DO UPDATE set col_2 = 55 RETURNING *;
COMMIT;
```

AND

```SQL
BEGIN;
 DELETE FROM test_ref_table;
 insert into source_table_1  SELECT col_2, col_1 from target_table ON CONFLICT (col_1) DO UPDATE set col_2 = 55 RETURNING *;
COMMIT;
```

You should probably add primary key to source_table_1, or potentially create another table if that fails some of the tests.